azure-ipam/go.mod

```diff
@@ -0,0 +1,82 @@
+module github.com/Azure/azure-container-networking/azure-ipam
```

Comment:
Could you try running a `go mod tidy` from within the `azure-ipam` directory? When I tried it, I got this error:
```
go: finding module for package github.com/containernetworking/cni/pkg/types/current
github.com/Azure/azure-container-networking/azure-ipam imports
        github.com/Azure/azure-container-networking/cns/client imports
        github.com/Azure/azure-container-networking/cns/restserver imports
        github.com/Azure/azure-container-networking/cns/ipamclient imports
        github.com/Azure/azure-container-networking/ipam tested by
        github.com/Azure/azure-container-networking/ipam.test imports
        github.com/containernetworking/cni/pkg/types/current: module github.com/containernetworking/cni@latest found (v1.1.1), but does not contain package github.com/containernetworking/cni/pkg/types/current
```

Actually, do we even need `azure-ipam` to be a go module? It seems like there's already a go.mod at the repository root.

Reply:
I want it to be a separate module; I've been trying to move the functionally distinct binaries in this monorepo to modules [for a while](https://github.com/Azure/azure-container-networking/pull/1136).
As for your build error - multimodule workspaces are new in go1.18 and with nested modules require that you have a [`go.work`](https://go.dev/doc/tutorial/workspaces) in the root module. (there is a `make workspace` target that _should_ set this up but I wouldn't rely on it yet.